### PR TITLE
Add unset_field and unset_fields to Event

### PR DESCRIPTION
**Pull request by lan17**
Wrote unit tests.

They pass.

### CODE REVIEW

**Comment by lan17** [comment]
👓  @bsimpson63 @spladug

**Comment by spladug** [comment]
The tests are failing on the spellchecker:

```
baseplate/events (line   1) param 
baseplate/events (line   1) param 
baseplate/events (line   1) iterable 
```

(you can ignore the spam about various packages not being installed, that's just the documentation generator complaining).

File: baseplate/events/queue.py [hunk 1]
```diff
@@ -80,6 +80,25 @@ def set_field(self, key, value, obfuscate=False):
...
             obfuscated_payload[key] = value
 
+    def unset_field(self, key):
+        """Remove the key from payload if it exists.
+        :param key:
+        :return: value of key that was removed.  None otherwise
+        """
+        return self.payload.pop(key, None)
+
+    def unset_fields(self, keys):
```
Comment (spladug) [inline baseplate/events/queue.py]: I'm not really sure we need this function, if a caller wants to do this they can loop themselves.

File: baseplate/events/queue.py [hunk 2]
```diff
@@ -80,6 +80,25 @@ def set_field(self, key, value, obfuscate=False):
             obfuscated_payload = self.payload.setdefault("obfuscated_data", {})
             obfuscated_payload[key] = value
 
+    def unset_field(self, key):
+        """Remove the key from payload if it exists.
+        :param key:
```
Comment (spladug) [inline baseplate/events/queue.py]: I think what's going on with the spellchecker is that rST is pretty picky about line spacing (and even if it weren't it'd be good to be consistent with other docstrings in this package). The param should also actually be documented. How about the following for docs for consistency sake:

```python
"""Remove a field from the event if it exists.

:param str key: The name of the field.
:return: The value of the removed field or :py:data:`None`.

"""
```

this function should also probably support the `obfuscated` flag like `get_field` does.

File: tests/unit/events/queue_tests.py [hunk 3]
```diff
@@ -77,6 +77,38 @@ def test_serialize(self, time, uuid):
             },
         })
 
+    def test_unset_field(self):
+        event = Event("topic", "type")
+        event.set_field("foo", "bar")
+
+        # foo should point to bar
```
Comment (spladug) [inline tests/unit/events/queue_tests.py]: :+1: :)

**Comment by spladug** [comment]
:nail_care: thanks!

File: baseplate/events/queue.py (hunk 1 above)
Comment (lan17) [inline baseplate/events/queue.py]: I was going to use it in my other PR for convenience.

File: baseplate/events/queue.py (hunk 2 above)
Comment (lan17) [inline baseplate/events/queue.py]: fixed

File: tests/unit/events/queue_tests.py (hunk 3 above)
Comment (lan17) [inline tests/unit/events/queue_tests.py]: added more tests for obfuscated.

File: baseplate/events/queue.py (hunk 1 above)
Comment (lan17) [inline baseplate/events/queue.py]: removed from this PR.

**Comment by lan17** [comment]
💇 
fixed the typos!

**Comment by spladug** [comment]
going to close this for now as discussed in slack.

**Pull request closed without merging**